### PR TITLE
More site routes

### PR DESCRIPTION
**Pull request by florisdobber**
Tickets: https://code4community-team.monday.com/boards/925293934/pulses/1185019867
https://code4community-team.monday.com/boards/925293934/pulses/1218648886

I will rename functions and adjust db tables in a separate PR to minimize confusion.

### CODE REVIEW

File: service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java [hunk 1]
```diff
@@ -59,6 +66,19 @@ public void unfavoriteSite(JWTData userData, int siteId) {
...
 
+  @Override
+  public FavoriteSitesResponse getFavoriteSites(JWTData userData) {
+    List<FavoriteSitesRecord> favoriteSites =
+        db.selectFrom(FAVORITE_SITES)
+            .where(FAVORITE_SITES.USER_ID.eq(userData.getUserId()))
+            .fetch();
+    List<Integer> favorites = new ArrayList<>();
+
+    favoriteSites.forEach(site -> favorites.add(site.getSiteId()));
```
Comment (connernilsen) [inline service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java:77]: Here you can probably do something like 
```java
List<Integer> favorites = db.select(FAVORITE_SITES.SITE_ID)
    .from(FAVORITE_SITES).where(...).fetch();
```
or
```java
List<Integer> favoriteSites =
        db.selectFrom(FAVORITE_SITES)
            .where(FAVORITE_SITES.USER_ID.eq(userData.getUserId()))
            .fetch(FAVORITE_SITES.SITE_ID);
```
to remove the extra logic that loops through each record

File: service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java [hunk 2]
```diff
@@ -74,4 +94,45 @@ public void recordStewardship(
...
+              new StewardshipActivity(
+                  record.getId(),
+                  record.getUserId(),
+                  record.getPerformedOn(),
+                  record.getWatered(),
+                  record.getMulched(),
+                  record.getCleaned(),
+                  record.getWeeded());
+          activities.add(stewardshipActivity);
+        });
```
Comment (connernilsen) [inline service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java:134]: For this one, you can try `.fetchOneInto(StewardshipActivity.class)`. Depending on how the DTO was set up, you might also need to add a private empty constructor to the class, but we'll look at that if you're getting exceptions.

**Review by connernilsen — COMMENTED** [review]
Looks good, just a couple questions first.

File: service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java (hunk 1 above)
Comment (florisdobber) [reply service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java:77]: Ah good call, was just copying other code without thinking

File: service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java (hunk 2 above)
Comment (florisdobber) [reply service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java:134]: Can you explain this a little more? fetchOneInto seems to fetch only one record, but I want a list of records.

**Review by liammoyn — APPROVED** [review]

File: service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java (hunk 2 above)
Comment (connernilsen) [reply service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java:131]: Oh whoops, I think the method i should have said would be something like 
`.fetchInto(SomeDto.class)`

Comment (florisdobber) [reply service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java:131]: Is this worth it? Seems to just be confusing things when the current method works

**Review by connernilsen — APPROVED** [review]
Looks good

**Pull request merged**